src/fides/api/ctl/migrations/versions/643249f65453_case_insensitive_usernames_delete_.py

```diff
@@ -0,0 +1,48 @@
+"""case_insensitive_usernames_delete_duplicates
+
+Revision ID: 643249f65453
+Revises: 392992c7733a
+Create Date: 2023-02-01 22:22:38.055862
+
+"""
+from alembic import op
+from sqlalchemy.orm.session import Session
+from sqlalchemy.sql import text
+
+from fides.lib.models.fides_user import FidesUser
+from fides.lib.models.fides_user_permissions import FidesUserPermissions
+
+# revision identifiers, used by Alembic.
+revision = "643249f65453"
+down_revision = "392992c7733a"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+
+    # Found here in the 24.2.2.4. Nondeterministic Collations section:
+    # https://www.postgresql.org/docs/current/collation.html
+    op.execute(
+        "CREATE COLLATION IF NOT EXISTS case_insensitive (provider = icu, locale = 'und-u-ks-level2', deterministic = false);"
```

Comment:
This collation is needed it order to make the `DISTINCT ON` function work. Without it the DB will think every username is unique.

Reply:
You probably don't need a collation - something like upper should work?

```psql
SELECT DISTINCT ON (UPPER(username)) username, created_at, id
FROM fidesuser
ORDER BY UPPER(username), created_at ASC
```

Reply:
Interesting. I'll give that a try and report back